plugins/inputs/redis_sentinel/redis_sentinel.go

```diff
@@ -0,0 +1,488 @@
+		}
+
+		sentinel := redis.NewSentinelClient(
+			&redis.Options{
+				Addr:      address,
+				Password:  password,
+				Network:   u.Scheme,
+				PoolSize:  1,
+				TLSConfig: tlsConfig,
+			},
+		)
+
+		tags := map[string]string{}
+		if u.Scheme == "unix" {
+			tags["socket"] = u.Path
+		} else {
+			tags["source"] = u.Hostname()
+			tags["port"] = u.Port()
+		}
+
+		r.clients[i] = &RedisSentinelClient{
+			sentinel: sentinel,
+			tags:     tags,
+		}
+	}
+
+	r.initialized = true
+	return nil
+}
+
+// Redis list format has string key/values adjacent, so convert to a map for easier use
+func toMap(vals []interface{}) map[string]string {
+	m := make(map[string]string)
+
+	for idx := 0; idx < len(vals); idx += 2 {
+		m[vals[idx].(string)] = vals[idx+1].(string)
+	}
+
+	return m
+}
+
+// Reads stats from all configured servers accumulates stats.
+// Returns one of the errors encountered while gather stats (if any).
+func (r *RedisSentinel) Gather(acc telegraf.Accumulator) error {
+	if !r.initialized {
+		if err := r.init(acc); err != nil {
+			return err
+		}
+	}
+
+	var wg sync.WaitGroup
+
+	for _, client := range r.clients {
+		wg.Add(1)
+
+		go func(client *RedisSentinelClient, acc telegraf.Accumulator) {
+			defer wg.Done()
+
+			// First check all masters this sentinel is monitoring
+			mastersCmd := redis.NewSliceCmd("sentinel", "masters")
+			if smErr := client.sentinel.Process(mastersCmd); smErr != nil {
+				acc.AddError(smErr)
+				return
+			}
+
+			masters, mastersErr := mastersCmd.Result()
+			if mastersErr != nil {
+				acc.AddError(mastersErr)
+				return
+			}
+
+			for _, master := range masters {
+				m := toMap(master.([]interface{}))
+
+				quorumCmd := redis.NewStringCmd("sentinel", "ckquorum", m["name"])
+				if qErr := client.sentinel.Process(quorumCmd); qErr != nil {
+					acc.AddError(qErr)
+					return
+				}
+
+				_, quorumErr := quorumCmd.Result()
+
+				sentinelMastersTags, sentinelMastersFields := convertSentinelMastersOutput(client.BaseTags(), m, quorumErr)
+				acc.AddFields(measurementMasters, sentinelMastersFields, sentinelMastersTags)
+
+				// ------------------------------------------------------------
+
+				// Check other Sentinels
+				sentinelsCmd := redis.NewSliceCmd("sentinel", "sentinels", m["name"])
+				if ssErr := client.sentinel.Process(sentinelsCmd); ssErr != nil {
+					acc.AddError(ssErr)
+					return
+				}
+
+				sentinels, sentinelsErr := sentinelsCmd.Result()
+				if sentinelsErr != nil {
+					acc.AddError(sentinelsErr)
+					return
+				}
+
+				for _, sentinel := range sentinels {
+					sm := toMap(sentinel.([]interface{}))
+
+					sentinelTags, sentinelFields := convertSentinelSentinelsOutput(client.BaseTags(), m["name"], sm)
+					acc.AddFields(measurementSentinels, sentinelFields, sentinelTags)
+				}
+
+				// ------------------------------------------------------------
+
+				// Check other Replicas
+				replicasCmd := redis.NewSliceCmd("sentinel", "replicas", m["name"])
```

Comment:
I found that in versions of Redis < 5, this command is called `sentinel slaves <master-group-name>`.  If we make this call with the older name, the plugin will work with at least back to Redis 3.0, and it continues to work current version.